csfieldguide/static/interactives/clicking-with-shaking/js/clicking-with-shaking.js

```diff
@@ -0,0 +1,573 @@
+/**
+ * The datatable to show the results.
+ * @type {null}
+ */
+let table = null;
+
+/**
+ * How many times the user has missed the target for the current set.
+ * @type {number}
+ */
+let misses = 0;
+
+/**
+ * How many times the user has hit the target for the current set.
+ * @type {number}
+ */
+let hits = 0;
+
+/**
+ * The user's current score.
+ *
+ * Hits increment the score while misses decrement the score. The score should not go below 0. The user has won when
+ * the score is equal to GOAL_SCORE.
+ * @type {number}
+ */
+let score = 0;
+
+/**
+ * The score the user needs to reach to pass the current set.
+ * @type {number}
+ */
+let GOAL_SCORE = 5;
+
+/**
+ * The button element that the user must click.
+ * @type {null}
+ */
+let target = null;
+
+/**
+ * The button element that starts/restarts/resumes the game when clicked.
+ * @type {null}
+ */
+let playButton = null;
+
+/**
+ * The button element that toggles whether sfx will be played.
+ * @type {null}
+ */
```

Comment:
JSDoc. I use it because I'm used to JavaDoc, which JSDoc is very similar to. It's nice because all the documentation can be exported as HTML, though this is probably not as useful for a standalone script like this, whereas it is a lot more useful for code that is reused in many places.